com.playeveryware.eos/Runtime/Core/Utility/FileSystemUtility.cs

```diff
@@ -478,15 +499,235 @@ public static string ReadAllText(string path)
+        /// <param name="createDirectory">
+        /// Whether or not to create the directory structure represented by the
+        /// filepath.
+        /// </param>
+        public static void WriteFile(string filePath, string content, bool createDirectory = true)
+        {
+            // Use discard in-order to call async function without awaiting.
+            _ = WriteFileInternal(filePath, content, createDirectory);
+        }
+
+        /// <summary>
+        /// Asynchronously writes the given file contents to the given filepath,
+        /// optionally creating the directory structure if it does not already
+        /// exist.
+        /// </summary>
+        /// <param name="filePath">The path to write to.</param>
+        /// <param name="content">The contents to write to the file.</param>
+        /// <param name="createDirectory">
+        /// Whether or not to create the directory structure represented by the
+        /// filepath.
+        /// </param>
+        /// <returns>Task</returns>
+        public static async Task WriteFileAsync(string filePath, string content, bool createDirectory = true)
+        {
+            await WriteFileInternal(filePath, content, createDirectory);
+        }
+
+        /// <summary>
+        /// Asynchronously writes the given file contents to the given filepath,
+        /// optionally creating the directory structure if it does not already
+        /// exist.
+        /// </summary>
+        /// <param name="filePath">The path to write to.</param>
+        /// <param name="content">The contents to write to the file.</param>
+        /// <param name="createDirectory">
+        /// Whether or not to create the directory structure represented by the
+        /// filepath.
+        /// </param>
+        /// <returns>Task</returns>
+        private static async Task WriteFileInternal(string filePath, string content, bool createDirectory = true)
+        {
+            FileInfo file = new(filePath);
+
+            // If the directory should be created, and the DirectoryInfo is not
+            // null, then create the directory.
+            if (createDirectory && null != file.Directory)
+            {
+                CreateDirectory(file.Directory);
+            }
+
+            await using StreamWriter writer = new(filePath);
+            await writer.WriteAsync(content);
+        }
+
+        /// <summary>
+        /// Helper function to create a directory. If the directory already
+        /// exists, then nothing will happen.
+        /// </summary>
+        /// <param name="dInfo">
+        /// The DirectoryInfo object that represents the directory to be
+        /// created.
+        /// </param>
+        private static void CreateDirectory(DirectoryInfo dInfo)
+        {
+            try
+            {
+                dInfo.Create();
+            }
+            catch(Exception ex) 
+            {
+                Debug.LogError($"Could not create directory " +
```

Comment:
Can we log the exception and pass it along?

Reply:
i.e. Debug.LogException

Reply:
Yuppers! All ficksed.